RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.h

```diff
@@ -0,0 +1,160 @@
+
+    /**
+     * Here we must implement the algorithm
+     */
+    std::vector<CSCSegment> run(const CSCChamber* aChamber, const ChamberHitContainer& rechits); 
+
+private:
+
+    /// Utility functions 
+    // Could be static at the moment, but in principle one
+    // might like CSCSegmentizer-specific behaviour?
+    bool areHitsCloseInR(const CSCRecHit2D* h1, const CSCRecHit2D* h2) const;
+    bool areHitsCloseInGlobalPhi(const CSCRecHit2D* h1, const CSCRecHit2D* h2) const;
+    bool isHitNearSegment(const CSCRecHit2D* h) const;
+
+    /**
+     * Try adding non-used hits to segment
+     */
+    void tryAddingHitsToSegment(const ChamberHitContainer& rechitsInChamber,
+	const BoolContainer& used, const LayerIndex& layerIndex,
+        const ChamberHitContainerCIt i1, const ChamberHitContainerCIt i2);
+
+    /**
+     * Return true if segment is 'good'.
+     * In this algorithm, 'good' means has sufficient hits
+     */
+    bool isSegmentGood(const ChamberHitContainer& rechitsInChamber) const;
+
+    /**
+     * Flag hits on segment as used
+     */
+    void flagHitsAsUsed(const ChamberHitContainer& rechitsInChamber, BoolContainer& used) const;
+	
+    /// Utility functions 	
+    bool addHit(const CSCRecHit2D* hit, int layer);
+    void updateParameters(void);
+    float fit_r_phi(SVector6 points, int layer) const;
+    float fitX(SVector6 points, SVector6 errors, float &chi2_str);
+    float fit_sp(SVector6 points, SVector6 errors, int layer);
+    float fitX_ir(SVector6 points, SVector6 errors, int ir, float &chi2_str);
+    float fitX_ir2(SVector6 points, SVector6 errors, int ir, int ir2, float &chi2_str);
+    void baseline(int n_seg_min);//function for arasing bad hits in case of bad chi2/NDOF 
+   /**
+     * Always enforce direction of segment to point from IP outwards
+     * (Incorrect for particles not coming from IP, of course.)
+     */
+    float phiAtZ(float z) const;
+    bool hasHitOnLayer(int layer) const;
+    bool replaceHit(const CSCRecHit2D* h, int layer);
+    void compareProtoSegment(const CSCRecHit2D* h, int layer);
+    void increaseProtoSegment(const CSCRecHit2D* h, int layer, int chi2_factor);
+
+		
+    // Member variables
+    // ================
+
+    const CSCChamber* theChamber;
+    ChamberHitContainer proto_segment;
+    const std::string myName; 
+		
+    double theChi2;
+    LocalPoint theOrigin;
+    LocalVector theDirection;
+    float uz, vz;
+    float windowScale;
+    int chi2D_iadd=1;
+    int strip_iadd=1;	
+    bool doCollisions;
+    float dRMax ;
+    float dPhiMax;
+    float dRIntMax;
+    float dPhiIntMax;
+    float chi2Max;
+    float chi2_str_;
+    float chi2Norm_2D_; 
+    float wideSeg;
+    int minLayersApart;
+    bool debugInfo;
+
+    CSCSegFit* sfit_;

```

Comment:
I am a bit confused. Using `std::unique_ptr<CSCSegFit>` in the header file, allows me to not care about cleaning `sfit_` in the entire source file? It is copied and reseted quite often.